deps.ffmpeg/40-aom.ps1

```diff
@@ -0,0 +1,102 @@
+param(
+    [string] $Name = 'aom',
+    [string] $Version = '3.6.1',
+    [string] $Uri = 'https://aomedia.googlesource.com/aom.git',
+    [string] $Hash = '7ade96172b95adc91a5d85bf80c90989cd543ee8',
+    [array] $FixupPatches = @(
+        @{
+            PatchFile = "${PSScriptRoot}/patches/aom/0001-windows-pkg-config-fix.patch"
+            HashSum = "22f38b49d6307c2ee860b08df7495b5f8894658b451c020f8a13162fd7dd29f4"
+        }
+    ),
+    [array] $Targets = @('x64')
+)
+
+function Setup {
+    Setup-Dependency -Uri $Uri -Hash $Hash -DestinationPath $Path
+
+    if ( ! ( $SkipAll -or $SkipDeps ) ) {
+        Invoke-External pacman.exe -S --noconfirm --needed --noprogressbar yasm
+    }
+}
+
+function Clean {
+    Set-Location $Path
+    if ( Test-Path "build_${Target}" ) {
+        Log-Information "Clean build directory (${Target})"
+        Remove-Item -Path "build_${Target}" -Recurse -Force
+    }
+}
+
+function Configure {
+    Log-Information "Configure (${Target})"
+    Set-Location $Path
+
+    $OnOff = @('OFF', 'ON')
+    $TargetCPUs = @{
+        x64 = 'x86_64'
+    }
+
+    $Options = @(
+        $CmakeOptions
+        "-DBUILD_SHARED_LIBS:BOOL=$($OnOff[$script:Shared.isPresent])"
+        '-DENABLE_DOCS:BOOL=OFF'
+        '-DENABLE_EXAMPLES:BOOL=OFF'
+        '-DENABLE_TESTDATA:BOOL=OFF'
+        '-DENABLE_TESTS:BOOL=OFF'
+        '-DENABLE_TOOLS:BOOL=OFF'
+        "-DAOM_TARGET_CPU=$($TargetCPUs[$Target])"
```

Comment:
Do we not also want:
```
-DENABLE_NASM:BOOL=ON
```

Reply:
It's omitted because it's supposed to use YASM.

Reply:
It can use either.
https://aomedia.googlesource.com/aom/+/refs/tags/v3.8.0/build/cmake/aom_config_defaults.cmake#188
https://aomedia.googlesource.com/aom/+/refs/tags/v3.8.0/build/cmake/aom_configure.cmake#157

While both projects are still active on GitHub, NASM tags/releases more frequently. The current Yasm tag/release (1.3.0) is from 2014, while NASM's current release (2.16.01) is from 2022.

We also currently set this option:
https://github.com/obsproject/obs-deps/blob/e1d4ddd710dbcc08407ff4c5c6267372b6bb0c83/deps.ffmpeg/40-aom.zsh#L63

Reply:
I haven't retried this recently (the PR is fairly old by this point) but if I were to hazard a guess then I might have tried building with NASM and ran into issues with that precisely for the same reason as the change above (mainly that building with MSVC doesn't seem well supported).

Reply:
```
    33 Warning(s)
    0 Error(s)
```
Seems like whatever might have been an issue isn't one anymore.